test/unittest/blas2/blas2_gbmv_test.cpp

```diff
@@ -0,0 +1,131 @@
+/***************************************************************************
+ *
+ *  @license
+ *  Copyright (C) Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  SYCL-BLAS: BLAS implementation using SYCL
+ *
+ *  @filename blas2_gbmv_test.cpp
+ *
+ **************************************************************************/
+
+#include "blas_test.hpp"
+
+template <typename T>
+using combination_t = std::tuple<int, int, int, int, T, T, bool, int, int, int>;
+
+template <typename scalar_t>
+void run_test(const combination_t<scalar_t> combi) {
+  index_t m;
+  index_t n;
+  index_t kl;
+  index_t ku;
+  bool trans;
+  scalar_t alpha;
+  scalar_t beta;
+  index_t incX;
+  index_t incY;
+  index_t lda_mul;
+  std::tie(m, n, kl, ku, alpha, beta, trans, incX, incY, lda_mul) = combi;
+
+  const char* t_str = trans ? "t" : "n";
```

Comment:
Shouldn't we test "c" as well. It is meant for complex but it should still work on real matrices.

Reply:
Thanks for your review, @fabiomestre!

Is the complex data type supported? I think that `c` stands for conjugate transpose and for real matrices it is equal to `t`.

Reply:
My understanding from the documentation is that 'c' will behave the same as 't' on  real matrices. So, even if we don't support complex yet, the API should still return correct results when using 'c' with real numbers.

https://spec.oneapi.io/versions/latest/elements/oneMKL/source/architecture/architecture.html#onemkl-datatypes

Reply:
The documentation about this was changed recently by https://github.com/oneapi-src/oneAPI-spec/pull/420

```diff
- -  Perform Hermitian transpose (transpose and conjugate). Only applicable to complex matrices.
+ -  Perform Hermitian transpose (transpose and conjugate). Is the same as ``transpose::trans`` for real matrices.
```

Reply:
I think we should try to match what the most recent oneAPI spec says.  And my understanding of the current code is that c should be working already for real numbers (the code defaults to the 't' case). So it's just a matter of updating the test.

If you think it's more work than that, then we can skip it.

Reply:
I confirm that `gbmv` acts as expected when invoked with `c` and real numbers.  For the moment, I suggest maintaining the `gbmv` test consistent with the others testing just `t` and `n`.